utilities.py

```diff
@@ -53,6 +53,48 @@ def classify_model():
     predout = pd.DataFrame({'id': test['id'], 'predicted': preds, 'actual': test['class']})
     return clf
 
+@logger_config.logger
+# classification into statements questions and chat with more choice of ml algorithms parameter tuned by Gridsearchcv.
+# By default this uses random forest if no argument passed.by @eaglewarrior
+def classify_model_adv(model="rf"):
+    import numpy as np
+    import pandas as pd
+    from sklearn.ensemble import RandomForestClassifier
+    from sklearn.svm import SVC
+    from xgboost import XGBClassifier
+    from sklearn.ensemble import AdaBoostClassifier
+    from sklearn.tree import DecisionTreeClassifier
+    from sklearn.naive_bayes import MultinomialNB
+    FNAME = Path('analysis/featuresDump.csv')
+    df = pd.read_csv(filepath_or_buffer=FNAME, )
+    df.columns = df.columns[:].str.strip()                                      # Strip any leading spaces from col names
+    df['class'] = df['class'].map(lambda x: x.strip())
+    width = df.shape[1]
+    # split into test and training (is_train: True / False col)
+    np.random.seed(seed=1)
+    df['is_train'] = np.random.uniform(0, 1, len(df)) <= .75
+    train, test = df[df['is_train'] == True], df[df['is_train'] == False]
+    features = df.columns[1:width-1]  # remove the first ID col and last col=classifier
+    # Fit an  Model for "class" given features rf:random forest,xgb:xgboost,nb:naive bayes,ada:adaboost
+    if model=="svm":
+	clf = SVC(C=1000, cache_size=200, class_weight=None, coef0=0.0,\
+	decision_function_shape='ovr', degree=3, gamma=0.0001, kernel='rbf',\
+	max_iter=-1, probability=False, random_state=None, shrinking=True,\
+	tol=0.001, verbose=False)
+    elif model=="rf":
+	clf= RandomForestClassifier(n_jobs=2, n_estimators=100)
+    elif model=="xgb":
+	clf= XGBClassifier(learning_rate= 0.01, n_estimators= 500)
+    elif model=="nb":
+	clf= MultinomialNB()
+    ## best performance by adaboost
+    elif model=="ada":
+	clf=AdaBoostClassifier(learning_rate=0.001,n_estimators=2000)
+    clf.fit(train[features], train['class'])
+    # Predict against test set
+    preds = clf.predict(test[features])
+    predout = pd.DataFrame({'id': test['id'], 'predicted': preds, 'actual': test['class']})
+    return clf
```

Comment:
@eaglewarrior This looks good to me. 
A couple of observations here:
1. I was thinking of a function which could run all models (say every month), compare them and then save the state of the current best. @vishakha-lall thoughts? 
This may or may not be picked up in the same issue. 
2. Since you've added this function, you also need to make changes where the old function is getting called.

Reply:
1. Yes it can be done with little changes in this function itself, but tell me on which metric it should decide that a particular algorithm is best to run , if that is fixed I can change the function and make it compatible to perform the task

2. This function by default will run old function only as by default it runs on randomforest

Reply:
1. Fair point, but I don't think it should be an issue even if it takes time once a month or once every two months. I think we can afford that cost for the improvement in responses.

2. Correct, but we've made this function configurable. The mail calls should point to this function now and potentially have a configurable command line parameter as well which can change the model being used. If I wish to change the function now, I can't change the parameter and do that as it still used the old one. As part of this issue, change the function calls to this function. Configuring a command-line parameter would be taken up in another issue.

Reply:
@janakrajchadha  I have raised PR #114  which is changes done as I have found the only chatbot.py is calling classify function just made changes in it . Please let me know if any issue.